cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamInputFormat.java

```diff
@@ -131,6 +133,16 @@ public static void setTimeRange(Configuration conf, long startTime, long endTime
   }
 
   /**
+   * Sets the stream id of the stream.
+   *
+   * @param conf The conf to modify.
+   * @param streamId {@link StreamId} id of the stream.
+   */
+  public static void setStreamId(Configuration conf, StreamId streamId) {
+    conf.set(STREAM_ID, streamId.toString());
```

Comment:
Oh I checked the code and it looks like the problem arises only for Datasets since we allow '.' in dataset IDs but we use '.' as a separator as well.